commit fb49f5f4560a6c4a28c213f4b474ab1321a3621f
Author: chinarulezzz <s.alex08@mail.ru>
Date:   Fri Mar 10 07:29:00 2017 +0200

    mequalizer plugin: fix tag removing if tag in equalizers combobox is equal to files tag

diff --git a/gmusicbrowser_layout.pm b/gmusicbrowser_layout.pm
index 9159a6f..3f35865 100644
--- a/gmusicbrowser_layout.pm
+++ b/gmusicbrowser_layout.pm
@@ -5423,6 +5423,15 @@ sub new
 	});
 	$mainbox->pack_start($_,0,0,0) for $turnoff, $save_preset;
 
+	::Watch($self, SongsChanged => sub {
+		if (Songs::Display($::SongID, 'version') && $::Options{equalizer_preset}
+		 && Songs::Display($::SongID, 'version') eq $::Options{equalizer_preset})
+		{   $save_preset->set_active(1);
+		}
+		else
+		{   $save_preset->set_active(0);
+		}
+	});
 	unless ($opt->{notoggle})
 	{	my $toggle= $self->{toggle}= Gtk2::ToggleButton->new;
 		$toggle->add(Gtk2::Image->new_from_stock('gtk-edit','menu'));
@@ -5485,7 +5494,9 @@ sub combo_changed_cb
 		return
 	}
 	::SetEqualizer(preset=>$current) if $current ne '';
-	$self->{save_preset}->set_active(0);
+	unless (Songs::Display($::SongID, 'version') eq $::Options{equalizer_preset})
+	{   $self->{save_preset}->set_active(0);
+	}
 	$::Options{PLUGIN_MEQUALIZER_EQU_PRESET} = $::Options{equalizer_preset};
 }
 sub button_cb
@@ -5507,12 +5518,12 @@ sub button_cb
 	}
 	elsif ($action eq 'turn_on')
 	{	::SetEqualizer(active=>1);
-		$self->{save_preset}->set_active(0);
+		$self->{save_preset}->set_active(0) unless $self->{save_preset}->get_active;
 		$::Options{PLUGIN_MEQUALIZER_EQU_AVAIL} = 1;
 	}
 	elsif ($action eq 'turn_off')
 	{	::SetEqualizer(active=>0);
-		$self->{save_preset}->set_active(0);
+		$self->{save_preset}->set_active(0) unless $self->{save_preset}->get_active;
 		$::Options{PLUGIN_MEQUALIZER_EQU_AVAIL} = 0;
 	}
 }
diff --git a/plugins/mequalizer.pm b/plugins/mequalizer.pm
index b73a8db..d77ad62 100644
--- a/plugins/mequalizer.pm
+++ b/plugins/mequalizer.pm
@@ -16,7 +16,6 @@ package GMB::Plugin::MEQUALIZER;
 use strict;
 use warnings;
 use constant OPT => 'PLUGIN_MEQUALIZER_';
-use Data::Dumper;
 
 my $handle;
 
@@ -50,7 +49,7 @@ sub Save {
                 Songs::Set($::SongID, version => "");
             }
         }
-    };
+    }
 }
 
 sub Start {
